extension/sql/timescale_prometheus.sql

```diff
@@ -0,0 +1,62 @@
+
+CREATE OR REPLACE FUNCTION _prom_catalog.const_support(internal) RETURNS INTERNAL
+AS '$libdir/timescale_prometheus_support', 'const_support'
+LANGUAGE C IMMUTABLE STRICT;
+
+--wrapper around jsonb_each_text to give a better row_estimate
+--for labels (10 not 100)
+CREATE OR REPLACE FUNCTION _prom_catalog.label_jsonb_each_text(js jsonb,  OUT key text, OUT value text)
+ RETURNS SETOF record
+ LANGUAGE internal
+ IMMUTABLE PARALLEL SAFE STRICT ROWS 10
+AS $function$jsonb_each_text$function$;
+
+--wrapper around unnest to give better row estimate (10 not 100)
+CREATE OR REPLACE FUNCTION _prom_catalog.label_unnest(label_array anyarray)
+ RETURNS SETOF anyelement
+ LANGUAGE internal
+ IMMUTABLE PARALLEL SAFE STRICT ROWS 10
+AS $function$array_unnest$function$;
+
+---------------------  comparison functions ---------------------
+
+CREATE OR REPLACE FUNCTION _prom_catalog.label_find_key_equal(label_key SCHEMA_PROM.label_key, pattern SCHEMA_PROM.pattern)
```

Comment:
instead of overwritting all these functions can we just have a noop stub for const_support in the connector sql and only overwrite that?

Reply:
unfortunately we can't; you're not allowed to use the `SUPPORT` option unless you're superuser